recipes/poco/all/conanfile.py

```diff
@@ -200,6 +202,9 @@ def generate(self):
         tc.preprocessor_definitions["POCO_NO_AUTOMATIC_LIBS"] = "1"
         # Picked up from conan v1 CMake wrapper, don't know the rationale
         tc.preprocessor_definitions["XML_DTD"] = "1"
+        # Enable debug and trace logs for Release builds
+        if self.options.log_debug:
+            tc.preprocessor_definitions["POCO_LOG_DEBUG"] = "1"
```

Comment:
```suggestion
```
It's not needed, because is only consumed by public headers. Only when consuming Poco it makes sense.

Reply:
Oh you are right!
I guess I can close this pull request and set this definition in my consumer project with `add_definitions(-DPOCO_LOG_DEBUG)`...

I was probably too fast creating that PR, because we fixed a similar issue with `POCO_NO_SHARED_LIBRARY_DEBUG_SUFFIX` and this definitely affects the generated library. So I've the following question: 

Should I abandon this pull request and create a new one for switching off `POCO_NO_SHARED_LIBRARY_DEBUG_SUFFIX`?
Or do you think it won't get merged anyways because it's not needed by most of the users to disable the debug-suffix?

Thanks for your help!

Reply:
You can keep this PR, but it should not affect the package ID at all. It can set a definition in cpp_info, like `self.cpp_info.defines.append('POCO_LOG_DEBUG=1')`

Reply:
Sorry for the late response, I've applied your suggestions!
The cpp_info is already there: `self.cpp_info.components["poco_foundation"].defines.append("POCO_LOG_DEBUG")`

Reply:
> I was probably too fast creating that PR, because we fixed a similar issue with `POCO_NO_SHARED_LIBRARY_DEBUG_SUFFIX` and this definitely affects the generated library. So I've the following question:
> 
> Should I abandon this pull request and create a new one for switching off `POCO_NO_SHARED_LIBRARY_DEBUG_SUFFIX`? Or do you think it won't get merged anyways because it's not needed by most of the users to disable the debug-suffix?

#22807
Now the `POCO_NO_SHARED_LIBRARY_DEBUG_SUFFIX` issue is also resolved!